### PR TITLE
WINC-1224: OCP 4.17 branching day changes

### PR DESCRIPTION
**Pull request by sebsoto**

### CODE REVIEW

**Review by jrvaldes — APPROVED** [review]
/lgtm

File: build/Dockerfile
Comment (saifshaikh48) [inline build/Dockerfile:1]: Do we also have to make this change in [ci-operator.yaml](https://github.com/openshift/windows-machine-config-operator/blob/master/.ci-operator.yaml#L5)?

File: .gitmodules [hunk 1]
```diff
@@ -15,27 +15,28 @@
 [submodule "kubelet"]
 	path = kubelet
 	url = https://github.com/openshift/kubernetes
-	branch = release-4.16
+	branch = release-4.17
 [submodule "kube-proxy"]
 	path = kube-proxy
 	url = https://github.com/openshift/kubernetes
 	branch = sdn-4.16-kubernetes-1.29.0
```
Comment (saifshaikh48) [inline .gitmodules:22]: Could you remind me when this is bumped/when we engage SDN to create the branch?

File: config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml
Comment (saifshaikh48) [inline config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml:1]: There are a few places where we reference 4.16 in the CSV description section that should be bumped as well. Curious if the pre-release script didn't work for this?

File: .gitmodules (hunk 1 above)
Comment (sebsoto) [reply .gitmodules:22]: It needs to be created after the kube rebase

File: build/Dockerfile
Comment (sebsoto) [reply build/Dockerfile:1]: You're correct, fixed

File: config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml
Comment (sebsoto) [reply config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml:1]: The script did not work

File: .gitmodules (hunk 1 above)
Comment (sebsoto) [reply .gitmodules:22]: The proposal was for the WINC team to maintain them :)

**Pull request merged**